### ISSUE TITLE
CustomImportOrder partially breaks ImportOrder when used in maven

### ISSUE THREAD

**Issue by elmuerte** [checkstyle/checkstyle#11380]
I've created an example project here: https://github.com/elmuerte/checkstyle-issue-importorder

When using the CustomImportOrder module and the ImportOrder module the following code is accepted when checked with maven.

```java
package example;

import org.springframework.boot.autoconfigure.cache.CacheProperties;

import org.springframework.util.CollectionUtils;
import org.yaml.snakeyaml.scanner.Constant;
```

When removing the CustomImportOrder module from the configuration the above code does not pass the checkstyle check via maven.

(The eclipse-cs plugin does report the violation.)

Expected result:

```
> mvn checkstyle:check
[INFO] Scanning for projects...
[INFO]
[INFO] ------------------< example:checkstyle-import-order >-------------------
[INFO] Building checkstyle-import-order 0.0.1-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- maven-checkstyle-plugin:3.1.2:check (default-cli) @ checkstyle-import-order ---
[INFO] Starting audit...
[ERROR] C:\dev\workspace\checkstyle-import-order\src\main\java\example\Foo.java:5:1: Extra separation in import group before 'org.springframework.util.CollectionUtils' [ImportOrder]
Audit done.
[INFO] There is 1 error reported by Checkstyle 10.0 with checkstyle-842.xml ruleset.
[ERROR] src\main\java\example\Foo.java:[5,1] (imports) ImportOrder: Extra separation in import group before 'org.springframework.util.CollectionUtils'
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.659 s
[INFO] Finished at: 2022-03-09T15:06:44+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.2:check (default-cli) on project checkstyle-import-order: You have 1 Checkstyle violation. -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

Actual result

```
> mvn checkstyle:check
[INFO] Scanning for projects...
[INFO]
[INFO] ------------------< example:checkstyle-import-order >-------------------
[INFO] Building checkstyle-import-order 0.0.1-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- maven-checkstyle-plugin:3.1.2:check (default-cli) @ checkstyle-import-order ---
[INFO] Starting audit...
Audit done.
[INFO] You have 0 Checkstyle violations.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.894 s
[INFO] Finished at: 2022-03-09T15:05:52+01:00
[INFO] ------------------------------------------------------------------------ 
```

**Comment by nick-mancuso** [comment]
@elmuerte thanks for example project. Please update issue description to follow template (with CLI output), this makes it easier for maintainers to evaluate issue and contributor/maintainer to clearly see failure and understand the problem.

**Comment by romani** [comment]
Issue will be closed at the end of March if no details provided

**Comment by elmuerte** [comment]
checkstyle-842.xml
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module PUBLIC "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN" "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <property name="severity" value="warning"/>
  <property name="charset" value="UTF-8"/>
  <property name="fileExtensions" value="java"/>
  <property name="tabWidth" value="4"/>
  <module name="TreeWalker">
    <module name="CustomImportOrder">
      <property name="severity" value="ignore"/>
      <property name="customImportOrderRules" value="STATIC###STANDARD_JAVA_PACKAGE###SPECIAL_IMPORTS###THIRD_PARTY_PACKAGE"/>
      <property name="thirdPartyPackageRegExp" value="^com\."/>
      <property name="specialImportsRegExp" value="^org\."/>
      <property name="sortImportsInGroupAlphabetically" value="true"/>
      <metadata name="net.sf.eclipsecs.core.lastEnabledSeverity" value="inherit"/>
    </module>
    <module name="ImportOrder">
      <property name="severity" value="error"/>
      <property name="option" value="top"/>
      <property name="groups" value="/^java\./,javax,org,com"/>
      <property name="separated" value="true"/>
    </module>
  </module>
</module>
```

src/main/java/example/Foo.java
```
package example;

import org.springframework.boot.autoconfigure.cache.CacheProperties;

import org.springframework.util.CollectionUtils;
import org.yaml.snakeyaml.scanner.Constant;

public class Foo {

        public Foo() {
                Class cls = Constant.class;
                CollectionUtils.arrayToList(null);
                new CacheProperties();
        }

}
```

CLI output
```
$ RUN_LOCALE="-Duser.language=en -Duser.country=US"
$ java $RUN_LOCALE -jar checkstyle-10.0-all.jar -c checkstyle-842.xml src/main/java/example/Foo.java
Starting audit...
[ERROR] /tmp/checkstyle-issue-importorder/src/main/java/example/Foo.java:5:1: Extra separation in import group before 'org.springframework.util.CollectionUtils' [ImportOrder]
Audit done.
Checkstyle ends with 1 errors.
```

yet, `mvn checkstyle:check` concludes "You have 0 Checkstyle violations." (as posted in description)

**Comment by strkkk** [comment]
@elmuerte what checkstyle version you use in maven plugin?

Please provide output of cli with version from maven.

**Comment by romani** [comment]
In post it is `checkstyle-10.0-all.jar`

**Comment by romani** [comment]
>When using the CustomImportOrder module and the ImportOrder module

What is a reason to use both ?

**Comment by romani** [comment]
>When using the CustomImportOrder module and the ImportOrder module

What is a reason to use both ?

>When removing the CustomImportOrder module from the configuration the above code does not pass

Checks are independent from each other, behavior like this means that maven plugin is misconfigured and might use default configuration, not your custom

**Comment by romani** [comment]
Please do problem statement completely based on CLI, if you can not reproduce problem by CLI, it means it is plugin problem and you need to report it to plugin issue tracker.

**Comment by elmuerte** [comment]
> @elmuerte what checkstyle version you use in maven plugin?

The issue happens with 8.40, 8.42, 9.0, and I reported it using 10.0.

>  What is a reason to use both ?

Not sure if there is a good reason. The CustomImportOrder configuration was originally used, and ImportOrder was added later. With the CustomImportOrder's severity reduced to "ignore".

> Please do problem statement completely based on CLI, if you can not reproduce problem by CLI, it means it is plugin problem and you need to report it to plugin issue tracker.

I can't reproduce it with just the CLI. So I will move this issue report to the maven plugin project.
